arch/arm/core/offsets/offsets_aarch32.c

```diff
@@ -83,17 +83,24 @@ GEN_OFFSET_SYM(_thread_stack_info_t, start);
  */
 #if defined(CONFIG_PM_S2RAM)
 GEN_OFFSET_SYM(_cpu_context_t, msp);
-GEN_OFFSET_SYM(_cpu_context_t, msplim);
 GEN_OFFSET_SYM(_cpu_context_t, psp);
-GEN_OFFSET_SYM(_cpu_context_t, psplim);
 GEN_OFFSET_SYM(_cpu_context_t, apsr);
 GEN_OFFSET_SYM(_cpu_context_t, ipsr);
 GEN_OFFSET_SYM(_cpu_context_t, epsr);
 
 GEN_OFFSET_SYM(_cpu_context_t, primask);
-GEN_OFFSET_SYM(_cpu_context_t, faultmask);
-GEN_OFFSET_SYM(_cpu_context_t, basepri);
 GEN_OFFSET_SYM(_cpu_context_t, control);
+
+#ifdef CONFIG_ARMV7_M_ARMV8_M_MAINLINE
+	/* Registers present on ARMv7-M and ARMv8-M */
+	GEN_OFFSET_SYM(_cpu_context_t, faultmask);
+	GEN_OFFSET_SYM(_cpu_context_t, basepri);
+#ifdef CONFIG_ARMV8_M_MAINLINE
+	/* Registers present only on ARMv8-M */
+	GEN_OFFSET_SYM(_cpu_context_t, msplim);
+	GEN_OFFSET_SYM(_cpu_context_t, psplim);
+#endif /* CONFIG_ARMV8_M_MAINLINE */
```

Comment:
These are also present in on Armv8-m baseline so support for these should be added too.